src/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java

```diff
@@ -452,6 +453,15 @@ public void setInputs( final Map<String, Object> inputs ) {
     if ( inputs.containsKey( DASHBOARD_MODE ) ) {
       dashboardMode = "true".equals( String.valueOf( inputs.get( DASHBOARD_MODE ) ) ); //$NON-NLS-1$
     }
+    if ( inputs.containsKey( QUERY_LIMIT ) ) {
+      try {
+        if ( this.getReport() != null && inputs.get( QUERY_LIMIT ) != null ) {
```

Comment:
getReport() never returns null. If no report-object is set, it will try to parse it from any of the other properties, and if that fails, it will throw an ResourceException.